### PR TITLE
Python2to3

### PR DESCRIPTION
**Pull request by jpoikela**
indents from tabs to spaces
convert with 2to3
compatibility bugs from conversion
- range needs int
- string encoding / byte sequence decoding
long print statements: indents fixed for the continuation

### CODE REVIEW

File: tools/map2header.py [hunk 1]
```diff
@@ -106,8 +109,8 @@ def compress_all(self):
 
     def room_split(self, buf_in, w, h):
         buf_out = []
-        for by in range(0, self.h/h):
-            for bx in range(0, self.w/w):
+        for by in range(0, int(self.h / h)):
```
Comment (jpoikela) [inline tools/map2header.py:112]: need to check if int() gives the correct value for this (also other places)

File: tools/map2header.py [hunk 2]
```diff
@@ -396,364 +411,371 @@ def dump_as_c_header(self, file, basename, global_properties):
...
+    def __init__(self, raw_map):
+        self.map_version = raw_map['version']
+        self.map_orientation = raw_map['orientation']
+        self.tile_w = raw_map['tilewidth']
+        self.tile_h = raw_map['tileheight']
+        self.map_w = raw_map['width']
+        self.map_h = raw_map['height']
+        self.raw_layers = raw_map['layers']
+        self.segment_w = self.map_w
+        self.segment_h = self.map_h
```
Comment (jpoikela) [inline tools/map2header.py:471]: uh, changing the indent makes the diff look bad

File: tools/map2header.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python3
```
Comment (jpoikela) [inline tools/map2header.py:1]: might be problematic if there are breaking changes in the future python3.x versions

File: tools/map2header.py (hunk 2 above)
Comment (jpoikela) [reply tools/map2header.py:471]: looks better in github diff when using "ignore whitespace changes"

**Comment by jpoikela** [comment]
there are some differences (looking into it now)
1. map_defs.h lines 6-11 have a space in the beginning (I'll try to find and fix it)
2. map_layer_objects.h has differences in the number of decimals: 15.6666666667 (py2.7) --> 15.6666666666666 (py3.9)

**Comment by jpoikela** [comment]
also in map_defs.h (but I think this is due to different dict iteration and does not change functionality)
`enum map_object_property_type {
    TYPE_PLANT, 
    TYPE_WATERDROP, 
    TYPE_ARCHER_SKELETON, 
    TYPE_SMILEY, 
    TYPE_SPIDER, 
};`
has become 
`enum map_object_property_type {
    TYPE_ARCHER_SKELETON, 
    TYPE_SPIDER, 
    TYPE_SMILEY, 
    TYPE_WATERDROP, 
    TYPE_PLANT, 
};`

**Comment by jpoikela** [comment]
the rounding adds some decimal places when compared to py2.7. do you think this is a problem?

**Comment by jpoikela** [comment]
the resulting rom zipped (on linux)
[game_test.rom.zip](https://github.com/retrodeluxe/rlengine-msx/files/5826797/game_test.rom.zip)

**Comment by jpoikela** [comment]
it's flawless on openmsx-catapult

**Comment by jpoikela** [comment]
one more thing: rlengine is awesome 😄

**Comment by retrodeluxe** [comment]
thanks

**Pull request merged**